### PR TITLE
fix(migrate): handle db creation errors & bump undici

### LINKED ISSUE #6426: Migrate leaks database password when Migration Engine binary crashes
## Bug description

Issue https://github.com/prisma/prisma/issues/6277 also exposes my database password, replaced with `foobar` below:

```
C:\Users\Jan\Documents\timescale>npx prisma migrate dev
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "foobar", schema "public" at "tsdb-2aa5e70e-prisma-0904.a.timescaledb.io:28514"

Error: Command failed with exit code 255: C:\Users\Jan\Documents\timescale\node_modules\@prisma\engines\migration-engine-windows.exe cli --datasource postgres://tsdbadmin:foobar@tsdb-2aa5e70e-prisma-0904.a.timescaledb.io:28514/foobar?sslmode=require create-database
Apr 06 09:57:15.665  INFO migration_engine: Starting migration engine CLI git_hash="60ba6551f29b17d7d6ce479e5733c70d9c00860e"
Apr 06 09:57:16.930 ERROR migration_engine::commands: Database error
Error querying the database: db error: FATAL: pg_hba.conf rejects connection for host "87.123.193.132", user "tsdbadmin", database "template1", SSL on
   0: sql_migration_connector::flavour::postgres::create_database
           with self=PostgresFlavour { features: BitFlags<MigrationFeature>(0b0), url: "<REDACTED>" }
             at migration-engine\connectors\sql-migration-connector\src\flavour\postgres.rs:116
   0: sql_migration_connector::flavour::postgres::create_database
           with self=PostgresFlavour { features: BitFlags<MigrationFeature>(0b0), url: "<REDACTED>" }
             at migration-engine\connectors\sql-migration-connector\src\flavour\postgres.rs:116
{"is_panic":false,"message":"Database error\nError querying the database: db error: FATAL: pg_hba.conf rejects connection for host \"87.123.193.132\", user \"tsdbadmin\", database \"template1\", SSL on\n   0: sql_migration_connector::flavour::postgres::create_database\n           with self=PostgresFlavour { features: BitFlags<MigrationFeature>(0b0), url: \"<REDACTED>\" }\n             at migration-engine\\connectors\\sql-migration-connector\\src\\flavour\\postgres.rs:116\n   0: sql_migration_connector::flavour::postgres::create_database\n           with self=PostgresFlavour { features: BitFlags<MigrationFeature>(0b0), url: \"<REDACTED>\" }\n             at migration-engine\\connectors\\sql-migration-connector\\src\\flavour\\postgres.rs:116","backtrace":null}
```

## How to reproduce

See https://github.com/prisma/prisma/issues/6277

## Environment & setup

```
C:\Users\Jan\Documents\timescale>npx prisma -v
Environment variables loaded from .env
prisma               : 2.20.1
@prisma/client       : 2.20.1
Current platform     : windows
Query Engine         : query-engine 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules\@prisma\engines\query-engine-windows.exe)
Migration Engine     : migration-engine-cli 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine : introspection-core 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary        : prisma-fmt 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash : 60ba6551f29b17d7d6ce479e5733c70d9c00860e
Studio               : 0.365.0
```

### PR DESCRIPTION
**Pull request by millsp**
closes #6426

### CODE REVIEW

**Comment by Jolg42** [comment]
Nice!

Note: also done in https://github.com/prisma/prisma/pull/6430

File: src/packages/cli/helpers/build.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ async function main() {
       target: 'node10',
       outfile: 'build/index.js',
       entryPoints: ['src/bin.ts'],
-      external: ['@prisma/engines'],
+      external: ['@prisma/engines', '_http_common'],
```
Comment (janpio) [inline src/packages/cli/helpers/build.js:26]: What is this about @millsp?

Comment (millsp) [reply src/packages/cli/helpers/build.js:26]: `undidci` uses a private Node.js module called `_http_common`

Comment (millsp) [reply src/packages/cli/helpers/build.js:26]: https://github.com/nodejs/undici/blob/v3.3.4/lib/node/http-parser.js

Comment (millsp) [reply src/packages/cli/helpers/build.js:26]: Seems to be removed in their `v4.0.0` release

Comment (Jolg42) [reply src/packages/cli/helpers/build.js:26]:
```
// TODO: This is not really allowed by Node but it works for now.
const common = require('_http_common')
```

Interesting 😄

**Pull request merged**